libs/common/src/test-helpers/schemas/custom-types-tests/with-entra-id-email-schema.tests.ts

```diff
@@ -0,0 +1,44 @@
+import { ZodSchema } from 'zod';
+import { WithSchemaTestParams } from '../types/with-schema-test.type';
+import { withDefaultOptionsTests } from '../primitive-types-tests/with-default-options.tests';
+
+export const withEntraIdEmailSchemaTests = <Schema extends ZodSchema>({
+  schema,
+  options = {},
+  getTestObjectWithUpdatedParameter,
+  getUpdatedParameterFromParsedTestObject,
+}: WithSchemaTestParams<Schema>) => {
+  describe('with withEntraIdEmailSchemaTests tests', () => {
+    withDefaultOptionsTests({
+      schema,
+      options,
+      getTestObjectWithUpdatedParameter,
+      getUpdatedParameterFromParsedTestObject,
+    });
+
+    it('should fail parsing if the parameter is not a null', () => {
```

Comment:
not a blocker, should this not read

```suggestion
    it('should fail parsing if the parameter is null', () => {
```
same for the rest